opensrp-chw/src/main/java/org/smartregister/chw/activity/AncHomeVisitActivity.java

```diff
@@ -105,7 +105,9 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
                         //refer
                         CoreReferralUtils.createReferralEvent(ChwApplication.getInstance().getContext().allSharedPreferences(),
                                 jsonString, "ec_anc_referral", baseEntityID);
-                        this.finish();
+                        if (buttonAction.equalsIgnoreCase("refer")){
```

Comment:
Is this necessary? You want to refer but continue with the visit?

Reply:
Hi @Andati ,

Yes we need them to continue with the flow after referral just like link to the ADDO shop but in this case it should be a link to the HF. When they have referred the ANC client who has never been to HF, they can come back and continue with other actions in that visit.